src/Compilers/Core/Portable/Collections/OrderPreservingMultiDictionary.cs

```diff
@@ -206,6 +219,37 @@ internal void Free()
                 }
             }
 
+            public bool TryGetValue<TArg>(Func<V, TArg, bool> predicate, TArg arg, [MaybeNullWhen(false)] out V value)
+            {
+                Debug.Assert(this.Count >= 1);
+                var arrayBuilder = _value as ArrayBuilder<V>;
+                if (arrayBuilder is not null)
+                {
+                    foreach (var v in arrayBuilder)
+                    {
+                        if (predicate(v, arg))
+                        {
+                            value = v;
+                            return true;
+                        }
+                    }
+
+                    value = default;
+                    return false;
```

Comment:
Consider moving after the `if` to avoid duplication.

Reply:
➡️ Amended commit to use this approach.